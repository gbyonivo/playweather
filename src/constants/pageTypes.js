export default {
  forecast: 'forecast',
  current: 'current'
};