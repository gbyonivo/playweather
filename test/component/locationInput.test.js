import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import LocationInput from '../../src/components/locationInput';
import combinedReducer from '../../src/reducers';
import pageTypes from '../../src/constants/pageTypes';

const props = {
  fetchData: () => { }
};

const state = {
  errorFetchingCurrentWeather: undefined,
  errorFetchingForecast: undefined,
  isFetchingCurrentWeather: false,
  isFetchingForecast: false,
};

describe('LocationInput component', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: state });
    const actual = shallow(
      <MemoryRouter initialEntries={[{ pathname: '/', match: { params: { type: pageTypes.current } } }]}>
        <Provider store={store}>
          <LocationInput {...props} />
        </Provider>
      </MemoryRouter>
    ).dive();
    expect(actual).toMatchSnapshot();
  });
});