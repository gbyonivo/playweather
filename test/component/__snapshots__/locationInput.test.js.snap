// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInput component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <withRouter(Connect(LocationInput))
    fetchData={[Function]}
  />
</Provider>
`;

exports[`LocationInput component renders correctly 2`] = `<div />`;
